print.py

```diff
@@ -0,0 +1,102 @@
+import sys
+import os
+import time
+from threading import Lock
+
+from ftxwebsocket.client import FtxWebsocketClient
+from ftxrest.client import FtxClient
+
+
+if len(sys.argv) < 2:
+    raise ValueError("Must specify a market")
+
+MARKET = sys.argv[1]
+
+last_printed_trade = None
+ws_trades = []
+print_lock = Lock()
+rest_client = FtxClient(
+    api_key="Doesnt matter, just using public APIs",
+    api_secret="Doesnt matter, just using public APIs",
+)
+
+
+def _fetch_missing_trades_when_appropriate():
+    last_trade = _get_last_ws_trade()
+
+    with print_lock:
+        while not _get_last_ws_trade() or _get_last_ws_trade() == last_trade:
+            time.sleep(.1)
+
+        _fetch_missing_trades()
+
+
+def _get_ws_trades(ws_client):
+    # Trades arrive in e.g. [[{T1}, {T2}, ...], [{T5}, {T6}, ...], ...] format,
+    # so flatten them
+    trade_lists = ws_client.get_trades(MARKET)
+    return [trade for trade_sublist in trade_lists for trade in trade_sublist]
+
+
+def _get_last_ws_trade():
+    if not ws_trades:
+        return None
+    else:
+        return ws_trades[-1]
+
+
+def _get_trade_time(trade):
+    from dateutil import parser
+    return parser.parse(trade["time"])
+
+
+def _sort_trades(trades):
+    # Sort by time and then by ID in case of equivalent times
+    sort_trade_key = lambda trade: (_get_trade_time(trade), int(trade["id"]))
+    return sorted(trades, key=sort_trade_key)
+
+
+def _fetch_missing_trades():
+    # Subtract one to account for upward timestamp rounding due to fractional seconds
+    start_time = int(_get_trade_time(last_printed_trade).timestamp()) - 1 if last_printed_trade else None
+    end_time = int(time.time())
+    trades_since_last = _sort_trades(rest_client.get_trades(market=MARKET, start_time=start_time, end_time=end_time))
```

Comment:
Note that this does not yet handle the case where more than 5000 (empirically observed request page size) trades occurred between websocket connection loss & restoration. We should ideally paginate based on `end_time`.